File: apple/Sources/ReactNativeWatchOS/Modifiers/CompositingGroup.swift
import SwiftUI

/// SwiftUI `.compositingGroup()`. Composites descendants as one layer
/// before effects (opacity, blend modes) apply.
enum RNWCompositingGroupModifier {
    @MainActor
    static func register(into r: RNWModifierRegistry) {
        r.register("compositingGroup") { view, _, _ in
            AnyView(view.compositingGroup())
        }
    }
}

File: src/modifiers/compositingGroup.ts
import { createModifier } from './createModifier';

/// SwiftUI `.compositingGroup()`. Composites the view's descendants as a
/// single layer before applying effects like opacity or blend modes, so
/// they apply to the group as a whole rather than each child.
export function compositingGroup() {
  return createModifier('compositingGroup');
}
